src/update.lua

```diff
@@ -8,47 +8,51 @@ local stringify = require(script.Parent.stringify)
 local currentVersion = '0.0.0' -- the current version of the plugin.
 
 local updateFunctions = {
-    none = function()
-        -- Fresh install / before 0.0.0
-        local uid = Plugin:GetSetting('user-id')
-        local uid2 = Plugin:GetSetting('UserId')
-        local tok = Plugin:GetSetting('refresh-token')
-        local ses = Plugin:GetSetting('session')
-        
-        local baseTable = {}
-        if uid or uid2 then
-            baseTable['user-id'] = uid or uid2
-            Plugin:SetSetting('user-id', nil)
-            Plugin:SetSetting('UserId', nil)
-        end
-        if tok then
-            baseTable['refresh-token'] = tok
-            Plugin:SetSetting('refresh-token', nil)
-        end
-        if ses then
-            baseTable['session'] = ses
-            Plugin:SetSetting('session', nil)
-        end
-        if baseTable ~= {} then
-            Plugin:SetSetting('auth', '!' .. stringify(baseTable))
-        else
-            Plugin:SetSetting('auth', '!{}')
-        end
-    end
+	none = function()
+		-- Fresh install / before 0.0.0
+		local uid = Plugin:GetSetting('user-id')
+		local uid2 = Plugin:GetSetting('UserId')
+		local tok = Plugin:GetSetting('refresh-token')
+		local ses = Plugin:GetSetting('session')
+		
+		local baseTable = {}
+		if uid or uid2 then
+			baseTable['user-id'] = uid or uid2
+			Plugin:SetSetting('user-id', nil)
+			Plugin:SetSetting('UserId', nil)
+		end
+		if tok then
+			baseTable['refresh-token'] = tok
+			Plugin:SetSetting('refresh-token', nil)
+		end
+		if ses then
+			baseTable['session'] = ses
+			Plugin:SetSetting('session', nil)
+		end
+		if baseTable ~= {} then
+			Plugin:SetSetting('auth', '!' .. stringify(baseTable))
+		else
+			Plugin:SetSetting('auth', '!{}')
+		end
+	end
 }
 
 local function update(pl)
-    Plugin = pl
-    local lastPluginVersion = Plugin:GetSetting('pluginVersion')
-    if lastPluginVersion ~= '#' .. currentVersion then
-        if lastPluginVersion == nil then
-            updateFunctions.none()
-        else
-            updateFunctions[lastPluginVersion]()
-        end
-        Plugin:SetSetting('pluginVersion', '#' .. currentVersion)
-        print('Successfully updated plugin values to version ' .. currentVersion .. '!')
-    end
+	Plugin = pl
+	local lastPluginVersion = Plugin:GetSetting('pluginVersion')
+	if lastPluginVersion ~= '#' .. currentVersion then
+		if lastPluginVersion == nil then
+			updateFunctions.none()
+		else
+			for version, func in ipairs(updateFunctions) do
```

Comment:
Ah, but we can't use ipairs either. Because ipairs only iterates over the integer keys. Perhaps we can use a list of updates and a map from a version to the index of the update in the list.